### PR TITLE
Added the 404 component

### LINKED ISSUE #275: Design a 404 Page for invalid URL paths
### **Description of the problem**
Design a 404 page for dealing with invalid URL paths, currently when a page is not found it simply redirects to a blank page with just DONUT navbar.

### **Prerequisites**
Adobe Illustrator/Photoshop
Vue.js
Basic JS

### **How to solve**
Sounds too complicated?Let’s focus in.
First of all, you should set up the project on your end and for that follow the steps mentioned here in [readme](https://github.com/codeuino/Social-Platform-Donut/blob/master/README.md) file.

Once you have successfully set up the project you are already halfway done👏🙌, so now coming to the actual problem and how to solve it. 
First, you need to design a cool/funky 404-page using illustrator or photoshop whatever you feel comfortable at and get it reviewed by mentors. Once the design is fixed you can start with the coding part for its frontend and create an individual (.vue) file for it and place it in /client/src/views folder.

Once the frontend part is ready, it's time to modify our routes for catching the fallback and redirect to our newly created 404 page. For adding new route modify the wild card(*) entry in server/server.js file, 

You can refer to these [amazing 404](https://optinmonster.com/best-404-page-examples/) pages for inspiration, but the design needs to be authentic and not copied from some other source.

### **Acceptance Criteria**
In order to successfully complete this task, you should be able to first design a cool(visually attractive) 404 page and once its design is confirmed by mentors you should be able to convert your design into code and finally modify the routes. 
If your submission is meeting all the above-mentioned requirements then woah you have successfully completed this task.

### **Rejection Criteria**
Don't worry if you get stuck somewhere or need some help feel free to contact the mentors through [slack channel](https://codeuino.slack.com/)

### PR DESCRIPTION
**Pull request by elit-altum**
Created a newly designed 404 page for handling unspecified routes.
Fixes #275 

File-animation-2

### CODE REVIEW

**Comment by jaskirat2000** [comment]
@devesh-verma can you take a look once. Thanks

File: src/router.js [hunk 1]
```diff
@@ -15,6 +16,7 @@ const Router = () => (
       <Route exact path="/pinned-posts" component={PinnedPosts} />
       <Route exact path="/profile" component={Profile} />
       <Route exact path="/organization" component={Organization} />
+      <Route component={NotFound} />
```
Comment (devesh-verma) [inline src/router.js]: path is missing for not found route that is 404

Comment (elit-altum) [reply src/router.js]: Sir no specific path is required there. The component 'NotFound' will render at every path not already specified above it. Therefore showing a 404 error for every route which is not specified in the app, and the user tries to access it.

Comment (devesh-verma) [reply src/router.js]: so usually, I have seen route as * for 404, please let me know if its not needed so that I can merge.

**Comment by devesh-verma** [comment]
@elit-altum can please resolve the merge conflict and update the PR.

**Pull request merged**